### PR TITLE
Text overflow handling and page finalize callback example

### PR DESCRIPTION
**Pull request by 3ace**

### CODE REVIEW

File: report/pdf_report.go [hunk 1]
```diff
@@ -119,6 +119,32 @@ func RunPdfReport(outputPath string) error {
 		block.Draw(p)
 	})
 
+	// Draw side note on each content page
+	c.PageFinalize(func(args creator.PageFinalizeFunctionArgs) error {
+		if args.PageNum < 3 {
+			return nil
+		}
+
+		p := c.NewStyledParagraph()
```
Comment (gunnsth) [inline report/pdf_report.go:128]: can you create a separate example for this perhaps?  it does not make the report look particularly nice.

File: text/pdf_formatted_text.go [hunk 2]
```diff
@@ -50,6 +50,12 @@ func main() {
 		log.Fatalf("Error: %v", err)
 	}
 
+	// Generate styled paragraph text wrapping and overflow subchapter.
+	err = styledParagraphTextOverflow(c, chap, fontRegular, fontBold)
```
Comment (adrg) [inline text/pdf_formatted_text.go:54]: There seems to be something wrong in this example (by looking at the output file). The content of the last chapter should fit in the available space but instead it is placed on a new one. I'm not sure what causes this (might be a bug regarding the margins of the chapter component). Does the whole thing fit on one page if you control the margins only from the paragraphs below the chapter/subchapter titles?

**Review by adrg — COMMENTED** [review]
The examples look good. However, the output of one of them seems wrong. 
I left a comment about that.

File: text/pdf_formatted_text.go (hunk 2 above)
Comment (gunnsth) [reply text/pdf_formatted_text.go:54]: Is that a bug in the example or unipdf?

Comment (3ace) [reply text/pdf_formatted_text.go:53]: It's a problem with the example, it's bottom margin is set to 20.

File: go.mod [hunk 3]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/boombuler/barcode v1.0.1
 	github.com/kr/text v0.2.0 // indirect
 	github.com/miekg/pkcs11 v1.0.3 // indirect
-	github.com/pkg/errors v0.9.1 // indirect
+	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.8.1 // indirect
 	github.com/unidoc/unipdf/v3 v3.23.0
```
Comment (gunnsth) [inline go.mod:11]: seems to be an old version?  should be v3.24.0

File: pages/pdf_page_side_note.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+import (
+	"fmt"
+	"log"
+
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/creator"
+	"github.com/unidoc/unipdf/v3/model"
+)
+
+const licenseKey = `
```
Comment (gunnsth) [inline pages/pdf_page_side_note.go:18]: use the api key as all other examples now using

**Review by gunnsth — APPROVED** [review]
LGTM

**Pull request merged**